File: app/src/main/java/com/shui520it/access/funcion/action/ShuiTextActionImpl.java
package com.shui520it.access.funcion.action;

import android.view.accessibility.AccessibilityNodeInfo;

import androidx.annotation.Nullable;

import java.util.ArrayList;
import java.util.List;

/**
 * @author shuimu{lwp}
 * @time 2019/8/14  11:59
 * @desc
 */
public class ShuiTextActionImpl extends ShuiBaseAction<String> {
    @Nullable
    @Override
    public AccessibilityNodeInfo findFirst(AccessibilityNodeInfo root) {
        List<AccessibilityNodeInfo> list = root.findAccessibilityNodeInfosByText(acitonKey);
        if (isEmpty(list)) {
            return null;
        }
        if (isEquals) {
            AccessibilityNodeInfo returnInfo = null;
            for (AccessibilityNodeInfo info : list) {
                if (returnInfo == null && info.getText() != null && acitonKey.equals(info.getText().toString())) {
                    returnInfo = info;
                } else {
                    info.recycle();
                }
            }
            return returnInfo;
        } else {
            return list.get(0);
        }
    }

    @Nullable
    @Override
    public List<AccessibilityNodeInfo> findAll(AccessibilityNodeInfo root) {
        List<AccessibilityNodeInfo> list = root.findAccessibilityNodeInfosByText(acitonKey);
        if (isEmpty(list)) {
            return null;
        }
        if (isEquals) {
            ArrayList<AccessibilityNodeInfo> listNew = new ArrayList<>();
            for (AccessibilityNodeInfo info : list) {
                if (info.getText() != null && acitonKey.equals(info.getText().toString())) {
                    listNew.add(info);
                } else {
                    info.recycle();
                }
            }
            return listNew;
        } else {
            return list;
        }
    }
}

File: app/src/main/java/com/shui520it/access/funcion/action/ShuiTextActionImpl.kt
package com.shui520it.access.funcion.action

import android.view.accessibility.AccessibilityNodeInfo

/**
 * @author shuimu{lwp}
 * @time  2021/2/1 14:17
 * @desc
 */
class ShuiTextActionImpl : ShuiBaseAction<String>() {
    override fun findFirst(info: AccessibilityNodeInfo): AccessibilityNodeInfo? {
        val findResultsByText = info.findAccessibilityNodeInfosByText(actionKey)
        return if (isEquals) {
            var b: Boolean
            findResultsByText?.find {
                b = it.text != null && actionKey.equals(it.text.toString())
                if (!b) it.recycle()
                b
            }
        } else
            findResultsByText?.get(0)
    }

    override fun findAll(info: AccessibilityNodeInfo): List<AccessibilityNodeInfo>? {
        val findResultByText = info.findAccessibilityNodeInfosByText(actionKey)
        return if (isEquals) {
            var b: Boolean
            findResultByText.filter {
                b = it.text != null && actionKey.equals(it.text.toString())
                if (!b) it.recycle()
                b
            }
        } else
            findResultByText
    }
}